cmd/finch-daemon/main.go

```diff
@@ -69,12 +74,34 @@ func runAdapter(cmd *cobra.Command, _ []string) error {
 	return run(options)
 }
 
+func handlePidFileOptions(options *DaemonOptions) error {
+	if options.PidFile != "" {
+		if err := os.MkdirAll(filepath.Dir(options.PidFile), 0o644); err != nil {
+			return fmt.Errorf("failed to create pidfile directory %s", err)
+		}
+		if err := pidfile.Write(options.PidFile, os.Getpid()); err != nil {
+			return fmt.Errorf("failed to start daemon, ensure finch daemon is not running or delete %s %s", options.PidFile, err)
```

Comment:
Error should be about failing to write pidfile

Reply:
this is from docker convention